pkg/volume/iscsi/iscsi.go

```diff
@@ -283,13 +306,74 @@ func portalMounter(portal string) string {
 	return portal
 }
 
-func getVolumeSource(spec *volume.Spec) (*v1.ISCSIVolumeSource, bool, error) {
+// get iSCSI volume info: readOnly and fstype
+func getISCSIVolumeInfo(spec *volume.Spec) (bool, string, error) {
 	if spec.Volume != nil && spec.Volume.ISCSI != nil {
-		return spec.Volume.ISCSI, spec.Volume.ISCSI.ReadOnly, nil
+		return spec.Volume.ISCSI.ReadOnly, spec.Volume.ISCSI.FSType, nil
 	} else if spec.PersistentVolume != nil &&
 		spec.PersistentVolume.Spec.ISCSI != nil {
-		return spec.PersistentVolume.Spec.ISCSI, spec.ReadOnly, nil
+		return spec.ReadOnly, spec.PersistentVolume.Spec.ISCSI.FSType, nil
+	}
+
+	return false, "", fmt.Errorf("Spec does not reference an ISCSI volume type")
+}
+
+// get iSCSI target info: target portal, portal, iqn, and lun
+func getISCSITargetInfo(spec *volume.Spec) (string, []string, string, int32, error) {
+	if spec.Volume != nil && spec.Volume.ISCSI != nil {
+		return spec.Volume.ISCSI.TargetPortal, spec.Volume.ISCSI.Portals, spec.Volume.ISCSI.IQN, spec.Volume.ISCSI.Lun, nil
+	} else if spec.PersistentVolume != nil &&
+		spec.PersistentVolume.Spec.ISCSI != nil {
+		return spec.PersistentVolume.Spec.ISCSI.TargetPortal, spec.PersistentVolume.Spec.ISCSI.Portals, spec.PersistentVolume.Spec.ISCSI.IQN, spec.PersistentVolume.Spec.ISCSI.Lun, nil
+	}
+
+	return "", nil, "", 0, fmt.Errorf("Spec does not reference an ISCSI volume type")
+}
+
+// get iSCSI inititator info: iface and initiator name
+func getISCSIInitiatorInfo(spec *volume.Spec) (string, *string, error) {
+	if spec.Volume != nil && spec.Volume.ISCSI != nil {
+		return spec.Volume.ISCSI.ISCSIInterface, spec.Volume.ISCSI.InitiatorName, nil
+	} else if spec.PersistentVolume != nil &&
+		spec.PersistentVolume.Spec.ISCSI != nil {
+		return spec.PersistentVolume.Spec.ISCSI.ISCSIInterface, spec.PersistentVolume.Spec.ISCSI.InitiatorName, nil
+	}
+
+	return "", nil, fmt.Errorf("Spec does not reference an ISCSI volume type")
+}
+
+// get iSCSI CHAP booleans
+func getISCSICHAPInfo(spec *volume.Spec) (bool, bool, error) {
+	if spec.Volume != nil && spec.Volume.ISCSI != nil {
+		return spec.Volume.ISCSI.DiscoveryCHAPAuth, spec.Volume.ISCSI.SessionCHAPAuth, nil
+	} else if spec.PersistentVolume != nil &&
+		spec.PersistentVolume.Spec.ISCSI != nil {
+		return spec.PersistentVolume.Spec.ISCSI.DiscoveryCHAPAuth, spec.PersistentVolume.Spec.ISCSI.SessionCHAPAuth, nil
+	}
+
+	return false, false, fmt.Errorf("Spec does not reference an ISCSI volume type")
+}
+
+// get iSCSI CHAP Secret info: secret namespace and secret name
+func getISCSISecretInfo(spec *volume.Spec, defaultSecretNamespace string) (string, string, error) {
```

Comment:
name this function to make it clear the order of returned variables, since both are strings. other volume helper methods used `getSecretNameAndNamespace()`, I think